ga4gh/backend.py

```diff
@@ -217,6 +219,62 @@ def variantSetsGenerator(self, request):
         return self._topLevelObjectGenerator(
             request, self._variantSetIdMap, self._variantSetIds)
 
+    def readsGenerator(self, request):
+        # Local utility function to save some typing
+        def getPosition(readAlignment):
+            return readAlignment.alignment.position.position
+
+        if len(request.readGroupIds) != 1:
+            raise exceptions.NotImplementedException(
+                "Read search over multiple readGroups not supported")
+        readGroupId = request.readGroupIds[0]
+        try:
+            readGroup = self._readGroupIdMap[request.readGroupIds[0]]
+        except KeyError:
+            raise exceptions.ReadGroupNotFoundException(readGroupId)
+        startPosition = request.start
+        equalPositionsToSkip = 0
+        if request.pageToken is not None:
+            startPosition, equalPositionsToSkip = self.parsePageToken(
+                request.pageToken, 2)
```

Comment:
Yes, very well spotted @macieksmuga! We should incorporate this check. We're going to need to think quite hard about how to test this --- I suspect a good start will be to make a read simulator which will generate all these nasty cases for us. This can be done after the beta though --- we're only promising initial support for reads in the beta release.